Updates to the Maintainers Guidelines and list of Maintainer Tasks (#80)
hyperledger/tsc: guidelines/SAMPLE-MAINTAINERS.md

> For information about the Maintainers of this repository, please see this Hyperledger \<PROJECT> repository's [MAINTAINERS](#) file.

## Maintainer Scopes, GitHub Roles and GitHub Teams

Maintainers are assigned the following scopes in this repository:

| Scope | Definition | GitHub Role | GitHub Team |
| ----- | ---------- | ----------- | ----------- |
| Maintainer | The GitHub Maintain role | Maintain   | `<repository> committers`     |

## Active Maintainers

<!-- Please keep this sorted alphabetically by github -->

| Name | GitHub ID | Scope | LFID | Discord ID | Email | Company Affiliation |
|----- | --------- | ----- | ---- | ---------- | ----- | ------------------- |
|      |           |       |      |            |       |                     |

## Emeritus Maintainers

| Name | GitHub ID | Scope | LFID | Discord ID | Email | Company Affiliation |
|----- | --------- | ----- | ---- | ---------- | ----- | ------------------- |
|      |           |       |      |            |       |                     |

## The Duties of a Maintainer

Maintainers are expected to perform the following duties for this repository. The duties are listed in more or less priority order:

> - Review, respond, and act on an reported security vulnerabilities reported against the repository.

Reviewer: ```suggestion
- Review, respond, and act on any reported security vulnerabilities reported against the repository.
```

This community welcomes contributions. Interested contributors are encouraged to
progress to become maintainers. To become a maintainer the following steps
occur, roughly in order.

- The proposed maintainer establishes their reputation in the community,
  including authoring five (5) significant merged pull requests, and expresses
  an interest in becoming a maintainer for the repository.
- A PR is created to update this file to add the proposed maintainer to the list of active maintainers.
- The PR is authored by an existing maintainer or has a comment on the PR from an existing maintainer supporting the proposal.
- The PR is authored by the proposed maintainer or has a comment on the PR from the proposed maintainer expressing their interest in being a maintainer.
  - The PR or comment from the proposed maintainer must include their
    willingness to be a long-term (more than 6 month) maintainer.
- Once the PR and necessary comments have been received, an approval timeframe begins.
- The PR **MUST** be communicated on all appropriate communication channels, including relevant community calls, chat channels and mailing lists. Comments of support from the community are welcome.

> - An add maintainer PR may be vetoed by another maintainer prior to merging.

Reviewer 2: I'm not sure I like veto power in the template. I'm aware that Besu has a defined veto, but do any other Hyperledger projects? Is it typical in open source projects?  It seems like veto gives a whole lot of power to a single individual...

Author: I took that from the example I had. I tried to clean up the language on it, but would be to soften it (objection, discussion, resolution) or remove it.  Perhaps easier to discuss?

Reviewer 3: It's not typical in other projects to my knowledge.  Some projects require supermajorities of maintainers to add another maintainer (2/3 or 3/4, for instance), but it's rare for a project to allow a single maintainer to block all other maintainers being added.

## Removing Maintainers

Being a maintainer is not a status symbol or a title to be carried
indefinitely. It will occasionally be necessary and appropriate to move a
maintainer to emeritus status. This can occur in the following situations:

- Resignation of a maintainer.
- Violation of the Code of Conduct warranting removal.
- Inactivity.
  - A general measure of inactivity will be no commits or code review comments
    for one reporting quarter. This will not be strictly enforced if
    the maintainer expresses a reasonable intent to continue contributing.
  - Reasonable exceptions to inactivity will be granted for known long term
    leave such as parental leave and medical leave.
- Other circumstances at the discretion of the other Maintainers.

The process to move a maintainer from active to emeritus status is comparable to the process for adding a maintainer, outlined above. In the case of voluntary
resignation, the Pull Request can be merged following a maintainer PR approval. If the removal is for any other reason, the following steps **SHOULD** be followed:

- A PR is created to update this file to move the maintainer to the list of emeritus maintainers.

> - The PR is authored by an existing maintainer or has a comment on the PR from an existing maintainer supporting the proposal.

Reviewer 2: ```suggestion
- The PR is authored by an existing maintainer or a Hyperledger staff member
```

Reviewer 2: Note - often times no existing maintainer wants to retire a colleague, it has been helpful to be able to ask a Hyperledger staff member to open a retirement PR. In fact, https://github.com/hyperledger/toc/blob/gh-pages/governing-documents/inactivity.md mentions it would be a TOC member or Hyperledger staff member, but I think it is acceptable for any project maintainer to do it.

Reviewer 4: Having done this many times by request, I agree with @denyeart

Author: I think adding “or a Hyperledger Organization GitHub Administrator” would make sense.  I’ll try that.
